source/apps/crawl_space/models.py

```diff
@@ -167,53 +161,36 @@ def get_solr_url(self):
     location = models.CharField(max_length=64, default="location")
     rounds_left = models.IntegerField(default=1, null=True, blank=True)
 
-    def __unicode__(self):
-        return self.name
-
     def save(self, *args, **kwargs):
-        # If this is the first time the model is saved, then the seeds
-        #    file needs to be moved from SEEDS_TMP_DIR/filename to the
-        #    crawl directory.
         if self.pk is None:
-            # Need to save first to obtain the pk attribute.
             self.slug = slugify(unicode(self.name))
             self.location = os.path.join(resources_dir, "crawls", self.slug)
-            super(Crawl, self).save(*args, **kwargs)
-
-            # Ensure that the crawl path `resources/crawls/<crawl.pk>` exists
+            # TODO:
+            # Fix this function and its weird side effect. Without this line the
+            # save method wont work.
             crawl_path = self.ensure_crawl_path()
-
-            # Move the file from temporary directory to crawl directory,
-            #   and update the FileField accordingly:
-            #   https://code.djangoproject.com/ticket/15590#comment:10
-
-            # Nutch requires a seed directory, not a seed file
-            if self.crawler == 'nutch':
-                seed_dir = join(crawl_path, 'seeds')
-                ensure_exists(seed_dir)
-                dst = join(crawl_path, 'seeds/seeds')
-                shutil.move(self.seeds_list.path, dst)
-                self.seeds_list.name = seed_dir
-            else:
-                dst = join(crawl_path, 'seeds')
-                shutil.move(self.seeds_list.path, dst)
-                self.seeds_list.name = dst
-                # Create unique configs for every ache crawl.
+            # If the crawler is ache, copy the config. If config already exists,
+            # delete it.
+            if self.crawler == 'ache':
                 if os.path.exists(self.get_config_path()):
                     shutil.rmtree(self.get_config_path())
                 shutil.copytree(self.get_default_config(), self.get_config_path())
                 self.config = self.get_config_path()
+        return super(Crawl, self).save(*args, **kwargs)
 
-
-            # Continue saving as normal
-
-        self.slug = slugify(unicode(self.name))
-        super(Crawl, self).save(*args, **kwargs)
-
+    # TODO:
+    # This is redundant. URL property is better.
```

Comment:
You could just hide this method by renaming it to _get_absolute_url().  I'm okay with keeping it as its own method.